community/kernel/src/test/java/org/neo4j/kernel/impl/query/Neo4jTransactionalContextTest.java

```diff
@@ -154,4 +154,78 @@ private void setUpMocks()
         when( dependencyResolver.resolveDependency( ThreadToStatementContextBridge.class ) ).thenReturn( statementContextBridge );
         when( dependencyResolver.resolveDependency( Guard.class ) ).thenReturn( guard );
     }
+
+    @SuppressWarnings( "ConstantConditions" )
+    @Test
+    public void rollsBackNewlyCreatedTransactionIfTerminationDetectedOnCloseDuringPeriodicCommit()
+    {
+        // Given
+        GraphDatabaseQueryService graph = mock( GraphDatabaseQueryService.class );
```

Comment:
i find this test extremely unreadable and unsupportable, only one thing you will do when it will fail - make mocks happy